webapp/channels/src/components/user_settings/plugin/pluginSetting.tsx

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useMemo, useState} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+import {get as getPreference} from 'mattermost-redux/selectors/entities/preferences';
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
+
+import SettingItemMax from 'components/setting_item_max';
+import SettingItemMin from 'components/setting_item_min';
+
+import type {GlobalState} from 'types/store';
+
+import RadioInput from './radio';
+import type {PluginConfigurationSetting} from './types';
+
+type Props = {
+    pluginId: string;
+    updateSection: (section: string) => void;
+    activeSection: string;
+    setting: PluginConfigurationSetting;
+}
+
+const PluginSetting = ({
+    pluginId,
+    setting,
+    activeSection,
+    updateSection,
+}: Props) => {
+    const dispatch = useDispatch();
+    const userId = useSelector(getCurrentUserId);
+    const preference = useSelector<GlobalState, string>((state: GlobalState) => getPreference(state, `pp_${pluginId}`, setting.name, setting.default));
+    const [selectedValue, setSelectedValue] = useState(preference);
+
+    const minDescribe = useMemo(() => {
+        if (setting.type === 'radio') {
+            return setting.options.find((v) => v.value === preference)?.text;
+        }
+
+        return undefined;
+    }, [setting, preference]);
+
+    const handleMinUpdateSection = (section: string): void => {
+        updateSection(section);
+    };
+
+    const updateSetting = async (value: string) => {
+        // Save preferences does not offer any await strategy or error handling
+        // so I am leaving this as is for now. We probably should update save
+        // preferences and handle any kind of error or network delay here.
+        dispatch(savePreferences(userId, [{
+            user_id: userId,
+            category: `pp_${pluginId}`.slice(0, 32),
+            name: setting.name,
+            value,
+        }]));
+        setting.onSubmit?.(setting.name, value);
+        updateSection('');
+    };
+
+    const inputs = [];
+    if (setting.type === 'radio') {
+        inputs.push(
+            <RadioInput
+                setting={setting}
+                selectedValue={selectedValue}
+                setSelectedValue={setSelectedValue}
+            />);
+    } else {
+        return null;
```

Comment:
This is nitpicking, but seeing a return only in an else statement seems odd to me. Would you mind breaking this up a bit by rewriting it as
```js
const inputs = [];
if (setting.type === 'radio') {
    inputs.push(...);
}

if (inputs.length === 0) {
    return null;
}

// ...
```
It's basically the same logic, but I think the flow is a bit easier to follow when one block just adds the inputs and the other checks if there's anything to render at all

Reply:
Far in the future, I expect this to be a switch/case, with the different types of settings. But for now I like the change you propose.